src/Runtime/Runtime.php

```diff
@@ -55,11 +55,11 @@ public function run($argv): int
     protected function doRun($argv, $output): int
     {
         // Do the preflight steps
-        $status = $this->preflight->preflight($argv);
+        $preflightSuccess = $this->preflight->preflight($argv);
 
         // If preflight signals that we are done, then exit early.
-        if ($status) {
-            return $status;
+        if ($preflightSuccess) {
```

Comment:
Here is the real bug. The original code was [if ($status !== false)](https://github.com/drush-ops/drush/commit/1a7704edfe99a7a223f65154a823318e2ca09499#diff-fcb54a74a331c6112bd5cf5e32f08a4d0a39f42c16833b899b4fac0339d453c7R219). That is, Preflight would return `bool false` if `doRun` should continue, and an integer status code if a redispatch happened. This sort of coding convention was common in the PHP 5 era, but is no longer considered de rigor in PHP 8, and in fact is a deprecated practice.

Reply:
Thanks for the feedback, this should be fixed now.